### ISSUE TITLE
spotifyd does terminates instantly without any error message

### ISSUE THREAD

**Issue by maringuu** [Spotifyd/spotifyd#549]
**Description**
Spotifyd exits _instantly_ without any error message and exit code 0.

Am I doing something wrong? I read the README and did not find any other docs.
Did I miss something?

**To Reproduce**
 Just start spotifyd with the foloowing config. Username and passwd obviously have to be changed to be valid.

```
[global]
# Your Spotify account name.
username = username

# Your Spotify account password.
password = password
backend = pulseaudio

device_name = device_name_in_spotify_connect

bitrate = 160

cache_path = /tmp/spotifyd

volume_normalisation = true
normalisation_pregain = -10

device_type = speaker
```

**Expected behavior**
You would expect an error message indicating what went wrong.

**Logs**
There are no logs.

**Compilation flags**
I use [spotifyd-full](https://aur.archlinux.org/packages/spotifyd-full/) from the AUR.
I am not sure, but I guess all flags are enabeled 
- [x] dbus_mpris
- [x] dbus_keyring
- [x] alsa_backend
- [x] portaudio_backend
- [x] pulseaudio_backend
- [x] rodio_backend

**Versions (please complete the following information):**
 - OS: ArchLinux
 - Spotifyd:0.2.24
- cargo: None

**Comment by 779g** [comment]
Had the same error on arch. First, I was using the alsa backend and a hw device, which worked alright until I wanted to change to the pulseaudio backend. 

I tried to switch to pulse with the same config as you and spotifyd daemon ran for a few seconds. When I selected the spotifyd device in spotify tui and started playing a song, It never started and I realized spotifyd daemon had crashed.

I tried uninstalling spotifyd-full-git and reinstalling spotifyd-pulseaudio-git, and nothing changed
